lib/autoscale/aws/discovery.go

```diff
@@ -46,15 +48,50 @@ func (a *Autoscaler) PublishDiscovery(ctx context.Context, operator ops.Operator
 		case <-publishTicker.C:
 			err = a.syncDiscovery(ctx, operator, false)
 			if err != nil {
-				a.Errorf("Failed to publish discovery: %v.", trace.DebugReport(err))
+				a.WithError(err).Error("Failed to publish discovery.")
 			}
 		case <-resyncTicker.C:
 			err = a.syncDiscovery(ctx, operator, true)
 			if err != nil {
-				a.Errorf("Failed to publish discovery: %v.", trace.DebugReport(err))
+				a.WithError(err).Error("Failed to publish discovery.")
 			}
+		case <-sourceDestTicker.C:
+			err = a.checkSourceDestination(ctx, operator)
+			if err != nil {
+				a.WithError(err).Error("Failed to ensure source/dest check is disabled.")
+			}
+		}
+	}
+}
+
+// checkSourceDestination makes sure source/destination check is disabled for
+// the cluster instances.
+func (a *Autoscaler) checkSourceDestination(ctx context.Context, operator ops.Operator) error {
+	cluster, err := operator.GetLocalSite(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	var instanceIDs []string
+	for _, node := range cluster.ClusterState.Servers {
+		if node.InstanceID != "" {
+			instanceIDs = append(instanceIDs, node.InstanceID)
+		} else {
+			a.Warnf("%v doesn't have cloud instance id.", node)
+		}
+	}
+	instances, err := a.DescribeInstancesWithSourceDestinationCheck(ctx, instanceIDs)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, instance := range instances {
+		a.Infof("Instance %v has source/dest check enabled, disabling.",
+			aws.StringValue(instance.InstanceId))
```

Comment:
Do you need to convert to an aws String for logging?

Reply:
Yeah that's what I'm basically doing here, `aws.String()` actually returns a string pointer. `InstanceId` field is a string pointer so `aws.StringValue()` dereferences it and returns a value.